cellprofiler/modules/watershed.py

```diff
@@ -566,31 +573,61 @@ def upgrade_settings(self, setting_values, variable_revision_number, module_name
 
         if variable_revision_number == 1:
             # Last two items were moved down to add more options for seeded watershed
-            __settings__ = setting_values[:-2]
+            new_values = setting_values[:-2]
 
-            # Add default connectivity and compactness
-            __settings__ += [1, 0.0]
+            new_values += [default_settings["connectivity"],
```

Comment:
Per previous convention, I thiiiink we want to leave the hardcoding here, but I'm willing to be argued out of it

Reply:
Same applies down the rest of the upgrade_settings